### PR TITLE
Added custom training container examples, based on the Amazon SageMaker training toolkit.

### PR DESCRIPTION
**Pull request by giuseppeporcelli**
*Issue #, if available:*

*Description of changes:*
This pull request contain four custom training container examples which show how to leverage on the recently released Amazon SageMaker training toolkit to enable script mode and loading user training modules from Amazon S3.

Additional documentation is in the readme.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sagemaker-custom-training-containers/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Amazon SageMaker Custom Training containers
+
+This folder contains skelethon implementations of Amazon SageMaker-compatible training containers.
```
Comment (ajaykarpur) [inline sagemaker-custom-training-containers/README.md:3]:
```suggestion
This folder contains skeleton implementations of Amazon SageMaker-compatible training containers.
```

File: sagemaker-custom-training-containers/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+The purpose of these examples is explaining how to build a custom container for training, with a relevant focus on the Amazon SageMaker Training Toolkit, a toolkit that facilitates the development of training containers for SageMaker, and enables dynamic loading of user scripts from Amazon S3, thus separating the execution environment (Docker container) from the script being executed. For additional info please see: [https://github.com/aws/sagemaker-training-toolkit](https://github.com/aws/sagemaker-training-toolkit).
```
Comment (ajaykarpur) [inline sagemaker-custom-training-containers/README.md:5]:
```suggestion
The purpose of these examples is to explain how to build a custom container for training using the Amazon SageMaker Training Toolkit. This toolkit facilitates the development of SageMaker-compatible training containers and enables dynamic loading of user scripts from Amazon S3, thus separating the execution environment (Docker container) from the script being executed. For additional info, please see: [https://github.com/aws/sagemaker-training-toolkit](https://github.com/aws/sagemaker-training-toolkit).
```

File: sagemaker-custom-training-containers/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+By purpose, no specific ML science is applied in this context, and code is simulating training dummy models.
```
Comment (ajaykarpur) [inline sagemaker-custom-training-containers/README.md:8]:
```suggestion
These examples deliberately do not use any specific ML framework or algorithm. The custom training container functionality is demonstrated by training dummy models.
```

File: sagemaker-custom-training-containers/basic-training-container/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# Part of the implementation of this container is based on the Amazon SageMaker Apache MXNet container.
+# https://github.com/aws/sagemaker-mxnet-container
+
+FROM ubuntu:16.04
+
+LABEL maintainer="Giuseppe A. Porcelli"
```
Comment (ajaykarpur) [inline sagemaker-custom-training-containers/basic-training-container/docker/Dockerfile:6]:
```suggestion
LABEL maintainer="Amazon AI"
```

File: sagemaker-custom-training-containers/framework-container/docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,64 @@
+# Part of the implementation of this container is based on the Amazon SageMaker Apache MXNet container.
+# https://github.com/aws/sagemaker-mxnet-container
+
+FROM ubuntu:16.04
+
+LABEL maintainer="Giuseppe A. Porcelli"
```
Comment (ajaykarpur) [inline sagemaker-custom-training-containers/framework-container/docker/Dockerfile:6]:
```suggestion
LABEL maintainer="Amazon AI"
```

File: sagemaker-custom-training-containers/script-mode-container-2/docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,61 @@
+# Part of the implementation of this container is based on the Amazon SageMaker Apache MXNet container.
+# https://github.com/aws/sagemaker-mxnet-container
+
+FROM ubuntu:16.04
+
+LABEL maintainer="Giuseppe A. Porcelli"
```
Comment (ajaykarpur) [inline sagemaker-custom-training-containers/script-mode-container-2/docker/Dockerfile:6]:
```suggestion
LABEL maintainer="Amazon AI"
```

File: sagemaker-custom-training-containers/script-mode-container/docker/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,67 @@
+# Part of the implementation of this container is based on the Amazon SageMaker Apache MXNet container.
+# https://github.com/aws/sagemaker-mxnet-container
+
+FROM ubuntu:16.04
+
+LABEL maintainer="Giuseppe A. Porcelli"
```
Comment (ajaykarpur) [inline sagemaker-custom-training-containers/script-mode-container/docker/Dockerfile:6]:
```suggestion
LABEL maintainer="Amazon AI"
```

**Comment by ajaykarpur** [comment]
Thanks @giuseppeporcelli. Would you be able to move this new directory (`sagemaker-custom-training-containers`) into the [advanced_functionality](https://github.com/awslabs/amazon-sagemaker-examples/tree/master/advanced_functionality) directory?

**Comment by giuseppeporcelli** [comment]
You should have all the changes committed. Thanks.

**Pull request merged**